recipes/mp-units/all/conanfile.py

```diff
@@ -12,58 +14,84 @@ class MPUnitsConan(ConanFile):
     license = "MIT"
     url = "https://github.com/conan-io/conan-center-index"
     settings = "os", "arch", "compiler", "build_type"
-    requires = (
-        "fmt/7.0.3",
-        "ms-gsl/3.1.0"
-    )
-    generators = "cmake"
+    options = {
+        "downcast_mode": ["off", "on", "auto"]
+    }
+    default_options = {
+        "downcast_mode": "on"
+    }
     no_copy_source = True
 
     @property
     def _source_subfolder(self):
         return "source_subfolder"
 
-    @property
-    def _build_subfolder(self):
-        return "build_subfolder"
+    def requirements(self):
+        compiler = self.settings.compiler
+        self.requires("fmt/7.1.3")
+        self.requires("gsl-lite/0.38.0")
+        if compiler == "clang" and compiler.libcxx == "libc++":
+            self.requires("range-v3/0.11.0")
 
-    def _validate_compiler_settings(self):
+    def validate(self):
         compiler = self.settings.compiler
         version = Version(self.settings.compiler.version)
         if compiler == "gcc":
             if version < "10.0":
                 raise ConanInvalidConfiguration("mp-units requires at least g++-10")
+        elif compiler == "clang":
+            if version < "12":
+                raise ConanInvalidConfiguration("mp-units requires at least clang++-12")
         elif compiler == "Visual Studio":
             if version < "16":
-                raise ConanInvalidConfiguration("mp-units requires at least MSVC 16.7")
+                raise ConanInvalidConfiguration("mp-units requires at least Visual Studio 16.9")
         else:
-            raise ConanInvalidConfiguration("mp-units is supported only by gcc and Visual Studio so far")
+            raise ConanInvalidConfiguration("Unsupported compiler")
         if compiler.get_safe("cppstd"):
             check_min_cppstd(self, "20")
 
-    def configure(self):
-        self._validate_compiler_settings()
-
     def source(self):
-        tools.get(**self.conan_data["sources"][self.version])
-        extracted_dir = "units-" + self.version
-        os.rename(extracted_dir, self._source_subfolder)
+        tools.get(**self.conan_data["sources"][self.version], strip_root=True, destination=self._source_subfolder)
+
+    def generate(self):
+        tc = CMakeToolchain(self)
+        tc.variables["UNITS_DOWNCAST_MODE"] = str(self.options.downcast_mode).upper()
+        tc.generate()
+        deps = CMakeDeps(self)
+        deps.generate()
```

Comment:
Do not use `CMakeToolchain` and `CMakeDeps` yet, they are not considered stable enough. Breaking changes can happen in the following Conan releases and they can break the recipe.

Reply:
Well, I must say I expected such an answer :wink:. Now the question is should I "oldfashionize" (the opposite term to "modernize" :wink:) and complicate my recipe or wait a bit until it will stabilize?

Reply:
I know 😄 . Waiting or _oldfashionizing_ is up to you. Stabilizing those generators might need a couple of releases at least (Conan client) and in ConanCenter we are usually one release behind the latest one... then we can start to consider it here. It is 3 months at least.

I would prefer to _oldfashionize_, maybe this recipe/version is useful for some other dependency. And afterward, we can come back and modernize.

Reply:
Done :-)